src/functions/api.js

```diff
@@ -15,145 +15,149 @@ const { ENDPOINT, CLOUDINARY_TESTDATAPATH } = require('./utils/config')
 
 const { db } = require('./utils/firebaseConfig')
 
+function startsWith(str, check) {
+  return str.indexOf(check) === 0
+}
+
+async function uploadDocument(res, name, base64) {
+  return cloudinary.uploader.upload(
+    base64,
+    {
+      public_id: CLOUDINARY_TESTDATAPATH + name,
+      resource_type: 'raw',
+      format: 'pdf',
+    },
+    (err, result) => {
+      if (err) {
+        console.error(
+          `Cloudinary was unable to upload the resume for "${name}" successfully`
+        )
+        console.log(err)
+        res.status(500).send('Unable to upload your resume to our CDN. Please try again')
+      }
+      console.log(`Cloudinary Successfully uploaded a resume for "${name}"`)
+      return result.secure_url
+    }
+  )
+}
+
 app.post(
   `${ENDPOINT}/api/file`,
   passport.authenticate('jwt', { session: false }),
   upload.none(),
   async (req, res) => {
-    /*TODO: -HANDLE ERROR INCASE CLOUDINARY UPLOAD DOESN'T WORK
-          -INPUT VALIDATION
-  */
-    if (req.user.verified === false) res.send(401)
-    //Upload the resume to the cloudinary CDN
-    let url = await cloudinary.uploader.upload(
-      req.body.pdf,
-      {
-        public_id: `${CLOUDINARY_TESTDATAPATH}${req.body.name} - Resume`,
-        resource_type: 'raw',
-        format: 'pdf',
-      },
-      (err, result) => {
-        if (err) {
-          console.error(
-            `Cloudinary was unable to upload the resume for ${req.body.name} successfully`
-          )
+    /* TODO:
+     *  - HANDLE ERROR INCASE CLOUDINARY UPLOAD DOESN'T WORK
+     *  - INPUT VALIDATION
+     *  - Select for majors
+     *  - Verification
+     */
+    if (!req.user.verified) {
+      res.sendStatus(401)
+      return
+    }
+
+    // Upload the resume to the cloudinary CDN
+    const url = await uploadDocument(res, `${req.body.name} - Resume`, req.body.pdf)
+
+    const profile = {
+      name: req.body.name,
+      linkedin: req.body.linkedin.toLowerCase(),
+      email: req.body.email,
+      gpa: req.body.gpa,
+      major: req.body.major,
+      standing: req.body.standing,
+      resume: url.secure_url,
+    }
+
+    if (!startsWith(profile.linkedin, 'https://www.linkedin.com/in/')) {
+      res.sendStatus(422)
+      return
+    }
```

Comment:
```suggestion
    const profile = {
      name: req.body.name,
      linkedin: req.body.linkedin.toLowerCase(),
      email: req.body.email,
      gpa: req.body.gpa,
      major: req.body.major,
      standing: req.body.standing,
      resume: url.secure_url,
    }
    
    if (!startsWith(profile.linkedin, 'https://www.linkedin.com/in/')) {
      res.status(422).send("The LinkedIn URL must start with: https://www.linkedin.com/in/")
    }
    
    // Upload the resume to the cloudinary CDN
    const url = await uploadDocument(res, `${req.body.name} - Resume`, req.body.pdf)
```

In the case that a user doesn't input a correctly formatted linkedin url or some other part of the input isn't correct, we wait until they've fixed that input to upload their resume.

Reply:
oop url wasnt defined